### DISCUSSION TITLE
What specific scenarios can this database be used in?

### DISCUSSION THREAD

**Discussion by Alexhuihui** [msiemens/tinydb#569] [General]
I took a look at the implementation, which uses full updates for writing and does not employ sequential file storage or ensure thread safety. It seems suitable only for educational purposes. I would like to know what production scenarios could utilize it.

**Comment by kinuax** [comment]
Scenarios without high performance and advanced features are feasible, as stated in the [Why Not Use TinyDB?](https://tinydb.readthedocs.io/en/latest/intro.html#why-not-use-tinydb) docs.

**Comment by ktdreyer** [comment]
My team and I needed a way to update a database in a Git repository and trigger actions based on the DB updates. TinyDB works for that. It it similar to SQLite, but the DB updates are nicely human-readable with `git` tooling. GitHub's web UI shows record additions, deletions, etc. My team needs transparency and approachability over raw performance at this stage of our application. We may outgrow TinyDB at some point, but we haven't hit that yet.

**Comment by msiemens** [comment]
I think this is better suited to a discussion, so I'm moving this issue 🙂

**Comment by msiemens** [comment]
I'd say TinyDB works best in situations where you have a small dataset that isn't big/complex enough to warrant using Sqlite and not worth the trouble of implementing or including some sort of schema migration mechanism. At least that's [how TinyDB got started in the first place](https://m-siemens.de/blog/2016/01/tinydb-past-present-and-future) 🙂
